plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -218,7 +218,10 @@ def _handle_yum_metadata_file(repo, type_id, unit_key, metadata, file_path, cond
 
     model = models.YumMetadataFile(**unit_data)
     model.set_storage_path(os.path.basename(file_path))
-    model.save_and_import_content(file_path)
+    try:
+        model.save_and_import_content(file_path)
+    except NotUniqueError:
+        model = model.__class__.objects.filter(**model.unit_key)
```

Comment:
This will return an iterator, which is probably not what you want here. I suggest using .get() or using .first() as you have used elsewhere.